src/Uno.UI/UI/Xaml/PropertyMetadata.cs

```diff
@@ -42,6 +42,14 @@ public partial class PropertyMetadata
 		/// </summary>
 		internal bool CoerceWhenUnchanged { get; set; } = true;
 
+		/// <summary>
+		/// Normally, when a property is updated and the coerced value is equal to
+		/// the value being updated to, the coerced value is thrown away. <seealso cref="DependencyObjectStore.ApplyCoercion"/>
+		/// This property is added specifically for Control.IsEnabledProperty in order
+		/// to override this behaviour.
+		/// </summary>
+		internal bool KeepCoercedWhenEquals { get; set; }
```

Comment:
I'd tend to think that moving this property to be a metadata option for FrameworkMetadata would be better:
https://github.com/unoplatform/uno/blob/86522f24d56fe39470def506e13fe6a7d85e352b/src/Uno.UI/UI/Xaml/FrameworkPropertyMetadataOptions.cs

Both CoerceWhenUnchanged and KeepCoercedWhenEquals are internal properties, which makes then appropriate to be in `FrameworkPropertyMetadataOptions`. It would also consume less memory (bit instead of a flag).

Can you try moving both as flags?